File: ios/Bulb.m
//
//  Bulb.m
//  RNLightApp
//
//  Created by Viet Pham on 2020-08-13.
//

#import "React/RCTBridgeModule.h"

@interface RCT_EXTERN_MODULE(Bulb, NSObject)
  
RCT_EXTERN_METHOD(turnOn)

RCT_EXTERN_METHOD(turnOff)

RCT_EXTERN_METHOD(getStatus: (RCTResponseSenderBlock)callback)

@end

File: ios/Bulb.swift
//
//  Bulb.swift
//  RNLightApp
//
//  Created by Viet Pham on 2020-08-13.
//

import Foundation

@objc(Bulb)
class Bulb: NSObject {
  
  @objc
  static var isOn = false
  
  @objc
  func turnOn() {
    Bulb.isOn = true
    print("Swift: Bulb is now ON")
  }
  
  @objc
  func turnOff() {
    Bulb.isOn = false
    print("Swift: Bulb is now OFF")
  }
  
  @objc
  func getStatus(_ callback: RCTResponseSenderBlock) {
    callback([NSNull(), Bulb.isOn])
  }
  
  @objc
  static func requiresMainQueueSetup() -> Bool {
    return true
  }
  
}
